src/med_bench/utils/classifiers.py

```diff
@@ -0,0 +1,508 @@
+        alphas = ALPHAS
+        cs = ALPHAS
+    else:
+        alphas = [TINY]
+        cs = [np.inf]
+
+    if not forest:
+        reg_y = RidgeCV(alphas=alphas, cv=CV_FOLDS)
+        clf_m = LogisticRegressionCV(Cs=cs, cv=CV_FOLDS)
+    else:
+        reg_y = RandomForestRegressor(n_estimators=100, min_samples_leaf=10)
+        clf_m = RandomForestClassifier(n_estimators=100, min_samples_leaf=10)
+    if calibration:
+        clf_m = CalibratedClassifierCV(clf_m, method=calib_method)
+
+    return reg_y, clf_m
+
+
+def _estimate_f_mu(t, m, x, y, crossfit, regressor_y, classifier_m, interaction):
+    """
+    Estimate mediator density f(M|T,X) and conditional mean outcome E[Y|T,M,X] with train
+    test lists from crossfitting
+
+    result has 2 outputs
+    - f  4-tuple of array-like, shape (n_samples) of densities/probas f(M=0|T=0,X), f(M=0|T=1,X), f(M=1|T=0,X),
+     f(M=1|T=1,X)
+    - mu 4-tuple of array-like, shape (n_samples) of conditional mean outcome E[Y|T=1,M=1,X], E[Y|T=1,M=0,X],
+    E[Y|T=0,M=1,X], E[Y|T=0,M=0,X]
+
+    """
+
+    n = len(y)
+    if len(x.shape) == 1:
+        x = x.reshape(-1, 1)
+    if len(m.shape) == 1:
+        mr = m.reshape(-1, 1)
+    else:
+        mr = np.copy(m)
+    if len(t.shape) == 1:
+        t = t.reshape(-1, 1)
+
+    t0 = np.zeros((n, 1))
+    t1 = np.ones((n, 1))
+    m0 = np.zeros((n, 1))
+    m1 = np.ones((n, 1))
+
+    train_test_list = _get_train_test_lists(crossfit, n, x)
+
+    mu_11x, mu_10x, mu_01x, mu_00x = [np.zeros(n) for _ in range(4)]
+    f_00x, f_01x, f_10x, f_11x = [np.zeros(n) for _ in range(4)]
+
+    x_t_mr = _get_interactions(interaction, x, t, mr)
+    t_x = _get_interactions(interaction, t, x)
+
+    x_t1_m1 = _get_interactions(interaction, x, t1, m1)
+    x_t1_m0 = _get_interactions(interaction, x, t1, m0)
+    x_t0_m1 = _get_interactions(interaction, x, t0, m1)
+    x_t0_m0 = _get_interactions(interaction, x, t0, m0)
+
+    t0_x = _get_interactions(interaction, t0, x)
+    t1_x = _get_interactions(interaction, t1, x)
+
+    for train_index, test_index in train_test_list:
+
+        reg_y = regressor_y.fit(x_t_mr[train_index, :], y[train_index])
+        clf_m = classifier_m.fit(t_x[train_index, :], m.ravel()[train_index])
+        mu_11x[test_index] = reg_y.predict(x_t1_m1[test_index, :])
+        mu_10x[test_index] = reg_y.predict(x_t1_m0[test_index, :])
+        mu_01x[test_index] = reg_y.predict(x_t0_m1[test_index, :])
+        mu_00x[test_index] = reg_y.predict(x_t0_m0[test_index, :])
+        f_00x[test_index] = clf_m.predict_proba(t0_x[test_index, :])[:, 0]
+        f_01x[test_index] = clf_m.predict_proba(t0_x[test_index, :])[:, 1]
+        f_10x[test_index] = clf_m.predict_proba(t1_x[test_index, :])[:, 0]
+        f_11x[test_index] = clf_m.predict_proba(t1_x[test_index, :])[:, 1]
+
+    f = f_00x, f_01x, f_10x, f_11x
+    mu = mu_11x, mu_10x, mu_01x, mu_00x
+    return f, mu
+
+def _get_y_m_x_classifiers(regularization, forest, calibration, calib_method):
```

Comment:
def _get_y_m_t_predictors instead?

Reply:
This has been addressed in 3f1b34577643b37b886202ab5873e7b5a9054a40